release-notes/10.0/preview/preview1/libraries.md

```diff
@@ -8,7 +8,173 @@
+
+Since SHA-2-256 ("SHA256") and SHA-3-256 have the same lengths,
+we decided that it wasn't pure goodness to let the Find method find any vaguely matching thumbprints.
+Instead, we introduced a new method that accepts the name of the hash algorithm that you want to use for matching.
+
+```C#
+X509Certificate2Collection coll = store.Certificates.FindByThumbprint(HashAlgorithmName.SHA256, thumbprint);
+Debug.Assert(coll.Count < 2, "Collection has too many matches, has SHA-2 been broken?");
+return coll.SingleOrDefault();
+```
+
+## Finding PEM-encoded Data in ASCII/UTF-8
+
+The PEM encoding (originally "Privacy Enhanced Mail", but now used widely outside of email) is defined for "text",
+which means that the `PemEncoding` class was designed to run on `System.String` and `ReadOnlySpan<char>`.
+But, it's quite common (especially on Linux) to have something like a certificate written in a file that uses the ASCII (string) encoding.
+Historically, that meant you needed to open the file, convert the bytes to chars (or a string), and then you can use PemEncoding.
+
+Taking advantage of the fact that PEM is only defined for 7-bit ASCII characters, and that 7-bit ASCII has a perfect overlap with single-byte UTF-8 values,
+you can now skip the UTF-8/ASCII to char conversion and read the file directly.
+
+```diff
+byte[] fileContents = File.ReadAllBytes(path);
+-char[] text = Encoding.ASCII.GetString(fileContents);
+-PemFields pemFields = PemEncoding.Find(text);
++PemFields pemFields = PemEncoding.FindUtf8(fileContents);
+
+-byte[] contents = Base64.DecodeFromChars(text.AsSpan()[pemFields.Base64Data]);
++byte[] contents = Base64.DecodeFromUtf8(fileContents.AsSpan()[pemFields.Base64Data]);
+```
+
+## New Method Overloads in ISOWeek for DateOnly Type
+
+The ISOWeek class was originally designed to work exclusively with DateTime, as it was introduced before the DateOnly type existed. Now that DateOnly is available, it makes sense for ISOWeek to support it as well.
+
+```C#
+    public static class ISOWeek
+    {
+        // New overloads
+        public static int GetWeekOfYear(DateOnly date);
+        public static int GetYear(DateOnly date);
+        public static DateOnly ToDateOnly(int year, int week, DayOfWeek dayOfWeek);
+    }
+```
+
+## String Normalization APIs to Work with Span of Characters
+
+Unicode string normalization has been supported for a long time, but existing APIs have only worked with the string type. This means that callers with data stored in different forms, such as character arrays or spans, must allocate a new string to use these APIs.
+Additionally, APIs that return a normalized string always allocate a new string to represent the normalized output.
+
+The change introduces new APIs that work with spans of characters, reducing the restriction to string types and helping to avoid unnecessary allocations.

```

Comment:
```suggestion
The change introduces new APIs that work with spans of characters, expanding normalization beyond string types and helping to avoid unnecessary allocations.

```